src/qt/optionsmodel.cpp

```diff
@@ -429,3 +429,21 @@ bool OptionsModel::isRestartRequired()
     QSettings settings;
     return settings.value("fRestartRequired", false).toBool();
 }
+
+void OptionsModel::checkAndMigrate()
+{
+    // Migration of default values
+    // Check if the QSettings container was already loaded with this client version
+    QSettings settings;
+    static const char strSettingsVersionKey[] = "nSettingsVersion";
+    if (!settings.contains(strSettingsVersionKey) || settings.value(strSettingsVersionKey) < CLIENT_VERSION)
```

Comment:
Just thought of this: we need to handle the case `!settings.contains(strSettingsVersionKey)` inside the if() as well.
```c++
int settingsVersion = settings.contains(strSettingsVersionKey) ? settings.value(strSettingsVersionKey) : 0;
if (settingsVersion < CLIENT_VERSION) {
    if (settingsVersion < 130000 && settings.contains("nDatabaseCache") && settings.value("nDatabaseCache") == 100)
    ....
}
```
Makes the code slightly cleaner as well